osu.Game/Screens/Edit/Timing/MetronomeDisplay.cs

```diff
@@ -0,0 +1,270 @@
+                            Size = triangleSize,
+                            Colour = overlayColourProvider.Background2,
+                            EdgeSmoothness = new Vector2(1),
+                            Alpha = 0.8f
+                        },
+                    }
+                },
+                bpmText = new OsuSpriteText
+                {
+                    Name = @"BPM display",
+                    Colour = overlayColourProvider.Content1,
+                    Anchor = Anchor.BottomCentre,
+                    Origin = Anchor.BottomCentre,
+                    Y = -3,
+                },
+            };
+        }
+
+        private double beatLength;
+
+        private TimingControlPoint timingPoint;
+
+        private bool isSwinging;
+
+        private readonly BindableInt interpolatedBpm = new BindableInt();
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            interpolatedBpm.BindValueChanged(bpm => bpmText.Text = bpm.NewValue.ToLocalisableString());
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (BeatSyncSource.ControlPoints == null || BeatSyncSource.Clock == null)
+                return;
+
+            timingPoint = BeatSyncSource.ControlPoints.TimingPointAt(BeatSyncSource.Clock.CurrentTime);
+
+            if (beatLength != timingPoint.BeatLength)
+            {
+                beatLength = timingPoint.BeatLength;
+
+                EarlyActivationMilliseconds = timingPoint.BeatLength / 2;
+
+                float bpmRatio = (float)Interpolation.ApplyEasing(Easing.OutQuad, Math.Clamp((timingPoint.BPM - 30) / 480, 0, 1));
+
+                weight.MoveToY((float)Interpolation.Lerp(0.1f, 0.83f, bpmRatio), 600, Easing.OutQuint);
+                this.TransformBindableTo(interpolatedBpm, (int)Math.Round(timingPoint.BPM), 600, Easing.OutQuint);
+            }
+
+            if (BeatSyncSource.Clock?.IsRunning != true && isSwinging)
+            {
+                swing.ClearTransforms(true);
+
+                using (swing.BeginDelayedSequence(350))
+                {
+                    swing.RotateTo(0, 1000, Easing.OutQuint);
+                    stick.FadeColour(overlayColourProvider.Colour2, 1000, Easing.OutQuint);
+                }
+
+                isSwinging = false;
+            }
+        }
+
+        protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, ChannelAmplitudes amplitudes)
+        {
+            base.OnNewBeat(beatIndex, timingPoint, effectPoint, amplitudes);
+
+            const float angle = 27.5f;
+
+            if (!IsBeatSyncedWithTrack)
+                return;
+
+            isSwinging = true;
+
+            float currentAngle = swing.Rotation;
+            float targetAngle = currentAngle > 0 ? -angle : angle;
+
+            swing.RotateTo(targetAngle, beatLength, Easing.InOutSine);
```

Comment:
Think this part isn't intentional? should be `InOutQuad` for the time being?

Reply:
oops.